### PR TITLE
feat: refactor contract detail data loading

### PR DESCRIPTION
**Pull request by songwongtp**
## Describe your changes
- Refactor contract detail data loading

### CODE REVIEW

File: src/lib/pages/account-details/index.tsx [hunk 1]
```diff
@@ -142,10 +142,7 @@ const AccountDetailsBody = ({ accountAddress }: AccountDetailsBodyProps) => {
         <TabPanels>
           <TabPanel p={0}>
             <TotalAccountValue accountAddress={accountAddress} />
-            <Flex
-              borderBottom={{ base: "0px", md: "1px solid" }}
-              borderBottomColor={{ base: "transparent", md: "gray.700" }}
-            >
+            <Flex borderBottom="1px solid" borderBottomColor="gray.700">
```
Comment (evilpeach) [inline src/lib/pages/account-details/index.tsx:145]: What about the mobile version? Is it still functioning properly?

File: src/lib/services/contractService.ts [hunk 2]
```diff
@@ -40,6 +46,70 @@ interface InstantiateDetail {
...
+        };
+      });
+  }, [contractAddress, indexerGraphClient]);
+
+  return useQuery(
+    ["contract_detail_by_contract", contractAddress, indexerGraphClient],
+    queryFn,
+    {
+      refetchOnWindowFocus: false,
+      enabled: !!contractAddress,
```
Comment (evilpeach) [inline src/lib/services/contractService.ts:76]: Boolean(contractAddress)

File: src/lib/services/contractService.ts [hunk 3]
```diff
@@ -40,6 +46,70 @@ interface InstantiateDetail {
...
+    return indexerGraphClient
+      .request(getContractByContractAddressQueryDocument, { contractAddress })
+      .then(({ contracts_by_pk }) => {
+        if (!contracts_by_pk) throw Error("Contract not found");
+        return {
+          contractAddress,
+          codeId: contracts_by_pk.code_id,
+          label: contracts_by_pk.label,
+          instantiator: contracts_by_pk.accountByInitBy?.address as Addr,
+          admin: contracts_by_pk.admin?.address as Addr,
```
Comment (evilpeach) [inline src/lib/services/contractService.ts:66]: I believe that the `admin` value could potentially return as null.

File: src/lib/services/contractService.ts [hunk 4]
```diff
@@ -40,6 +46,70 @@ interface InstantiateDetail {
...
+  contractAddress: ContractAddr,
+  onSuccess?: (data: ContractDetail) => void,
+  onError?: (err: Error) => void
+): UseQueryResult<ContractDetail> => {
+  const { indexerGraphClient } = useCelatoneApp();
+
+  const queryFn = useCallback(async () => {
+    return indexerGraphClient
+      .request(getContractByContractAddressQueryDocument, { contractAddress })
+      .then(({ contracts_by_pk }) => {
```
Comment (evilpeach) [inline src/lib/services/contractService.ts:59]: add `ContractDetail` as generic type on `then`

File: src/lib/services/contractService.ts [hunk 5]
```diff
@@ -40,6 +46,70 @@ interface InstantiateDetail {
...
+
+export const useInstantiateDetailByContractQuery = (
+  contractAddress: ContractAddr
+): UseQueryResult<InstantiateDetail> => {
+  const { indexerGraphClient } = useCelatoneApp();
+
+  const queryFn = useCallback(async () => {
+    return indexerGraphClient
+      .request(getInstantiateDetailByContractQueryDocument, { contractAddress })
+      .then(({ contracts_by_pk }) => ({
```
Comment (evilpeach) [inline src/lib/services/contractService.ts:91]: add `InstantiateDetail` as generic type on `then` as well

File: src/lib/services/contractService.ts [hunk 6]
```diff
@@ -40,6 +46,70 @@ interface InstantiateDetail {
...
+        initProposalTitle:
+          contracts_by_pk?.contract_proposals.at(0)?.proposal.title,
+      }));
+  }, [contractAddress, indexerGraphClient]);
+
+  return useQuery(
+    ["instantiate_detail_by_contract", contractAddress, indexerGraphClient],
+    queryFn,
+    {
+      enabled: !!contractAddress,
```
Comment (evilpeach) [inline src/lib/services/contractService.ts:108]: use `Boolean`

File: src/lib/query/contract.ts [hunk 7]
```diff
@@ -1,5 +1,56 @@
...
+              _in: [
+                "InstantiateContract"
+                "InstantiateContract2"
+                "SoftwareUpgrade"
+              ]
+            }
+          }
+        }
+        order_by: { proposal: { id: asc } }
+        limit: 1
```
Comment (evilpeach) [inline src/lib/query/contract.ts:43]: Could you also talk with the back-end team to ensure the correctness of proposal query?

File: src/lib/pages/contract-details/data.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+  const { data: contractDetail, isLoading: isContractDetailLoading } =
+    useContractDetailByContractAddress(contractAddress);
+
+  const { assetInfos, isLoading: isAssetInfosLoading } = useAssetInfos();
+  const { data: contractBalances, isLoading: isContractBalancesLoading } =
+    useQuery(
+      ["query", "contractBalances", balancesApiRoute, contractAddress],
+      async () => queryContractBalances(balancesApiRoute, contractAddress),
+      { enabled: !!contractAddress, retry: false }
```
Comment (evilpeach) [inline src/lib/pages/contract-details/data.ts:46]: Boolean

File: src/lib/pages/contract-details/data.ts [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+  const { data: publicInfo } =
+    usePublicProjectByContractAddress(contractAddress);
+  const { data: publicInfoBySlug } = usePublicProjectBySlug(publicInfo?.slug);
+
+  const { data: contractCw2Info, isLoading: isContractCw2InfoLoading } =
+    useQuery(
+      ["query", "contract_cw2_info", lcdEndpoint, contractAddress],
+      async () => queryContractCw2Info(lcdEndpoint, contractAddress),
+      { enabled: !!contractAddress, retry: false }
```
Comment (evilpeach) [inline src/lib/pages/contract-details/data.ts:71]: Boolean

**Review by evilpeach — CHANGES_REQUESTED** [review]
Please also fix the conflict.

File: src/lib/query/contract.ts (hunk 7 above)
Comment (bkioshn) [reply src/lib/query/contract.ts:43]: this is correct

File: src/lib/pages/account-details/index.tsx (hunk 1 above)
Comment (bkioshn) [reply src/lib/pages/account-details/index.tsx:192]: Yep, works perfectly splendid

**Review by evilpeach — APPROVED** [review]
lgtm

**Pull request merged**